### PR TITLE
Add a setting to auto start PIE clients when launching on device

### PR DESCRIPTION
**Pull request by improbable-valentyn**
#### Description
When launching on device with local deployment connection flow selected, the local deployment will need a server connected to it in order for the device to start gameplay. By default, only the dedicated server will be started without the PIE clients, but an option is added to the GDK to also start PIE clients.
Engine PR: https://github.com/improbableio/UnrealEngine/pull/464

#### Primary reviewers
@m-samiec @joshuahuburn

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDKEditor/Public/SpatialGDKEditorSettings.h [hunk 1]
```diff
@@ -380,6 +380,10 @@ class SPATIALGDKEDITOR_API USpatialGDKEditorSettings : public UObject
 	UPROPERTY(EditAnywhere, config, Category = "Mobile", meta = (DisplayName = "Extra Command Line Arguments"))
 	FString MobileExtraCommandLineArgs;
 
+	/** If checked, PIE clients will be automatically started when launching on device and connecting to local deployment. */
```
Comment (MatthewSandfordImprobable) [inline SpatialGDK/Source/SpatialGDKEditor/Public/SpatialGDKEditorSettings.h:383]: on a device

File: SpatialGDK/Source/SpatialGDKEditor/Public/SpatialGDKEditorSettings.h [hunk 2]
```diff
@@ -380,6 +380,10 @@ class SPATIALGDKEDITOR_API USpatialGDKEditorSettings : public UObject
...
+	UPROPERTY(EditAnywhere, config, Category = "Mobile", meta = (DisplayName = "Start PIE Clients when launching on device with local deployment flow"))
```
Comment (MatthewSandfordImprobable) [inline SpatialGDK/Source/SpatialGDKEditor/Public/SpatialGDKEditorSettings.h:384]: on a device perhaps?

File: SpatialGDK/Source/SpatialGDKEditor/Public/SpatialGDKEditorSettings.h [hunk 3]
```diff
@@ -380,6 +380,10 @@ class SPATIALGDKEDITOR_API USpatialGDKEditorSettings : public UObject
...
+	bool bStartPIEClientsWithLocalLaunchOnDevice = false;
```
Comment (MatthewSandfordImprobable) [inline SpatialGDK/Source/SpatialGDKEditor/Public/SpatialGDKEditorSettings.h:385]: Nit: Think we initialize settings in the constructor

File: SpatialGDK/Source/SpatialGDKEditor/Public/SpatialGDKEditorSettings.h (hunk 1 above)
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDKEditor/Public/SpatialGDKEditorSettings.h:383]:
```suggestion
	/** If checked, PIE clients will be automatically started when launching on a device and connecting to local deployment. */
```

File: SpatialGDK/Source/SpatialGDKEditor/Public/SpatialGDKEditorSettings.h (hunk 2 above)
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDKEditor/Public/SpatialGDKEditorSettings.h:384]:
```suggestion
	UPROPERTY(EditAnywhere, config, Category = "Mobile", meta = (DisplayName = "Start PIE Clients when launching on a device with local deployment flow"))
```

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorModule.cpp [hunk 4]
```diff
@@ -52,6 +52,11 @@ FString FSpatialGDKEditorModule::GetSpatialOSLocalDeploymentIP() const
 	return GetDefault<USpatialGDKEditorSettings>()->ExposedRuntimeIP;
 }
 
+bool FSpatialGDKEditorModule::ShouldStartPIEClientsWithLocalLaunchOnDevice() const
```
Comment (joshuahuburn) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorModule.cpp:55]: `LocalLaunch` here being spatial yeah?
Might get confusing in the future with `Launch` meaning `launch on device` with unreal. 
Perhaps we should say `LocalDeployment` ?

Comment (improbable-valentyn) [reply SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorModule.cpp:55]: It's a combination of the two in this case, `Launch on device` running with `Local deployment` flow selected for Spatial.

**Pull request merged**